haystack/components/builders/dynamic_prompt_builder.py

```diff
@@ -12,6 +12,10 @@
 logger = logging.getLogger(__name__)
 
 
+class ProcessedMessagesDict(TypedDict):
+    prompt: Any
```

Comment:
Shouldn't be like this?

```suggestion
    prompt: str
```